ql/src/java/org/apache/hadoop/hive/ql/exec/vector/VectorMapOperator.java

```diff
@@ -885,20 +885,7 @@ public void process(Writable value) throws HiveException {
             batchCounter++;
             oneRootOperator.process(deserializerBatch, 0);
 
-            /**
-             * Only reset the current data columns.  Not any data columns defaulted to NULL
-             * because they are not present in the partition, and not partition columns.
-             */
-            for (int c = 0; c < currentDataColumnCount; c++) {
-              ColumnVector colVector = deserializerBatch.cols[c];
-              if (colVector != null) {
-                colVector.reset();
-                colVector.init();
-              }
-            }
-            deserializerBatch.selectedInUse = false;
-            deserializerBatch.size = 0;
-            deserializerBatch.endOfFile = false;
+            deserializerBatch.reset();
```

Comment:
After reset, can you reinit isNull and isRepeating for columns which are not present in currentDataColumnCount? 
(i.e similar to the case mentioned in https://github.com/apache/hive/blob/master/ql/src/java/org/apache/hadoop/hive/ql/exec/vector/VectorMapOperator.java#L706)

Reply:
I excluded partition column and rowidentifiercolumn from reset. So this is more like the original code now. It only resets the output columns in addition to original implementation.